Added changelog for v.3.4.0 (#109)
MapsIndoors/mapsindoors.github.io: collections/_android/v3/CHANGELOG.md

### Security     in case of vulnerabilities.
-->

## [3.4.0] - 2020-03-18

### Added

> - Feature: Ability to override a Location's display rule at runtime (in version 2 this was achieved by using `Location.setType( type )`):

Reviewer: ```suggestion
- Feature: Ability to override a Location's display rule at runtime (in v2 this was achieved by using `Location.setType( type )`):
```

>   - `MapControl.setDisplayRule( displayRule, location | locations )`: Sets a display rule on the given location(s) that overrides the current ones being used.

Reviewer: ```suggestion
  - `MapControl.setDisplayRule( displayRule, location | locations )`: Sets a display rule on the given Location(s) that overrides the current ones being used.
```

>   - `MapControl.setDisplayRule( displayRuleName, location | locations )`: Sets display rule (by its type - name) on the given location(s) that overrides the current ones being used.

Reviewer: ```suggestion
  - `MapControl.setDisplayRule( displayRuleName, location | locations )`: Sets display rule (by its type - name) on the given Location(s) that overrides the current ones being used.
```

>   - `MapControl.getDisplayRule( location )`: Retrieves the current display rule being used by the given location.

Reviewer: ```suggestion
  - `MapControl.getDisplayRule( location )`: Retrieves the current display rule being used by the given Location.
```

>   - `MapControl.resetDisplayRule( location | locations )`: Removes the override display rule on the given location(s).

Reviewer: ```suggestion
  - `MapControl.resetDisplayRule( location | locations )`: Removes the override display rule on the given Location(s).
```

> - `MPFilter.Builder.setIgnoreLocationActiveStatus( ignoreActiveToFromStatus )`: Makes queries return location-results that are marked as inactive using the active from/to mechanism.

Reviewer: ```suggestion
- `MPFilter.Builder.setIgnoreLocationActiveStatus( ignoreActiveToFromStatus )`: Makes queries return Location that are marked as inactive using the "active from"/"active to" mechanism.
```

- `MPApiKeyValidatorService`: added a couple of new methods:

>   - `checkAPIKeyValidity( apiKey, listener )`: Checks the validity of the given MapsIndoors API Key. The SDK doesn't need to be initialized in order to call this method.

Reviewer: ```suggestion
  - `checkAPIKeyValidity( apiKey, listener )`: Checks the validity of the given MapsIndoors API key. The SDK doesn't need to be initialized in order to call this method.
```

>   - `validateAPIKeyStringFormat( apiKey )`: Checks that the given API Key string format is a valid one. This method only checks the string format.

Reviewer: ```suggestion
  - `validateAPIKeyStringFormat( apiKey )`: Checks that the given API key string format is a valid one. This method only checks the string format.
```

> - Feature: RoomId has been replaced by the ExternalId. The following methods have been added:

Reviewer: ```suggestion
- Feature: `RoomId` has been replaced by `ExternalId`. The following methods have been added:
```

>   - `MPLocation.getExternalId()`: Retrieves the externalId property, if available. Should be used instead of the now deprecated `MPLocation.getRoomId()`.

Reviewer: ```suggestion
  - `MPLocation.getExternalId()`: Retrieves the `externalId` property, if available. Should be used instead of the now deprecated `MPLocation.getRoomId()`.
```

>   - `MPLocation.Builder.setExternalId( externalId )`: Sets a custom externalId (a db item id, etc.). This replaces `MPLocation.Builder.setRoomId( roomId )`.

Reviewer: ```suggestion
  - `MPLocation.Builder.setExternalId( externalId )`: Sets a custom `externalId` (a db item id, etc.). This replaces `MPLocation.Builder.setRoomId( roomId )`.
```
